src/net/sourceforge/kolmafia/Modifiers.java

```diff
@@ -1295,7 +1295,7 @@ public static final int findBooleanName(String name) {
   private final double[] doubles;
   private final int[] bitmaps;
   private final String[] strings;
-  private ModifierExpression[] expressions;
+  private ArrayList<Indexed<ModifierExpression>> expressions;
```

Comment:
Why is this an improvement? Was it a sparse array to begin with?

Reply:
Right now it's a dense array, almost all of the elements of which are null. We iterate over all of them looking for expressions to evaluate. Instead, we should store a store list of pairs of (index, expression), which is what this does.